python/paddle/distributed/checkpoint/load_state_dict.py

```diff
@@ -0,0 +1,496 @@
+        for (
+            tensor_key,
+            local_tensor_metadata,
+        ) in metadata.state_dict_metadata.items():
+            if tensor_key not in storage_state_dict_metadata:
+                storage_state_dict_metadata[tensor_key] = []
+            storage_state_dict_metadata[tensor_key] += local_tensor_metadata
+    read_items = []
+    logger.debug(f"storage_state_dict_metadata:{storage_state_dict_metadata}")
+    for tensor_key, val in state_dict.items():
+        if isinstance(val, paddle.Tensor):
+            if val.is_dist():
+                (
+                    local_shape,
+                    global_offset,
+                ) = compute_local_shape_and_global_offset(
+                    val.shape,
+                    val.dist_attr.process_mesh,
+                    val.dist_attr.dims_mapping,
+                )
+            else:
+                local_shape = val.shape
+                global_offset = [0] * len(val.shape)
+            if not local_shape or not global_offset:
+                continue
+            cur_chunk_metadata = LocalTensorMetadata(global_offset, local_shape)
+            assert (
+                tensor_key in storage_state_dict_metadata
+            ), f"tensor_key:{tensor_key} not found in storage_state_dict_metadata:{storage_state_dict_metadata}."
+            for storage_local_tensor_metadata in storage_state_dict_metadata[
+                tensor_key
+            ]:
+                if not_overlap(
+                    cur_chunk_metadata, storage_local_tensor_metadata
+                ):
+                    continue
+                cur_offsets, storage_offsets, lengths = compute_overlap(
+                    cur_chunk_metadata, storage_local_tensor_metadata
+                )
+                storage_local_tensor_index = LocalTensorIndex(
+                    tensor_key,
+                    tuple(storage_local_tensor_metadata.global_offset),
+                )
+                read_items.append(
+                    ReadItem(
+                        storage_local_tensor_index,
+                        paddle.distributed.get_rank(),
+                        tuple(cur_offsets),
+                        tuple(storage_offsets),
+                        tuple(lengths),
+                    )
+                )
+        else:
+            raise ValueError(
+                f"Only support paddle.Tensor., val type:{type(val)}"
+            )
+    global_read_items = []
+    tmp = []
+    if use_dist:
+        paddle.distributed.all_gather_object(tmp, read_items, process_group)
+    else:
+        tmp.append(read_items)
+    for items in tmp:
+        for item in items:
+            global_read_items.append(item)
+    return global_read_items
+
+
+def load_state_dict(
+    state_dict,
+    path,
+    process_group=None,
+    coordinator_rank=0,
+) -> None:
+    """
+    Load the state_dict inplace from a checkpoint path.
+    Args:
```

Comment:
```suggestion
    
    Args:
```
声明、参数..等各部分之间加空行，否则可能会导致官网渲染出错